hcl/eval/eval.go

```diff
@@ -111,12 +108,44 @@ func (c *Context) PartialEval(expr hclsyntax.Expression) (hclwrite.Tokens, error
 	return engine.Eval()
 }
 
-// GetExpressionTokens gets the provided expression writable tokens.
-func GetExpressionTokens(hcldoc []byte, expr hclsyntax.Expression) (hclwrite.Tokens, error) {
-	exprRange := expr.Range()
+// TokensForValue returns the tokens for the provided value.
+func TokensForValue(value cty.Value) (hclwrite.Tokens, error) {
+	value, marks := value.Unmark()
+	if value.Type() == customdecode.ExpressionClosureType {
+		closureExpr := value.EncapsulatedValue().(*customdecode.ExpressionClosure)
+		for m := range marks {
+			switch v := m.(type) {
+			case ExpressionStringMark:
+				exprRange := closureExpr.Expression.Range()
+				exprData := []byte(v)[exprRange.Start.Byte:exprRange.End.Byte]
```

Comment:
Nice way to circumvent the lack of original bytes from the expression :100: (which keeps haunting us :laughing: )